frontend/src/pages/destructionlist/hooks.tsx

```diff
@@ -78,22 +91,69 @@ export function useDataGridProps(
     );
   }, []);
 
-  const fields = getFields(searchParams, zaaktypeChoicesState).map((field) => {
-    const isActiveFromStorage = fieldSelectionState?.[field.name];
-    const isActive =
-      typeof isActiveFromStorage === "undefined"
-        ? field.active !== false
-        : isActiveFromStorage;
-    return { ...field, active: isActive } as TypedField;
-  });
+  const getSpecificZaakSelection = async (url: string) => {
+    const zaakSelection = await getZaakSelection(storageKey);
+    if (!zaakSelection[url].selected) return;
+    return zaakSelection[url].detail;
+  };
+
+  const hasActions = Boolean(actions?.length);
+  const fields = getFields(searchParams, zaaktypeChoicesState, hasActions).map(
+    (field) => {
+      const isActiveFromStorage = fieldSelectionState?.[field.name];
+      const isActive =
+        typeof isActiveFromStorage === "undefined"
+          ? field.active !== false
+          : isActiveFromStorage;
+      return { ...field, active: isActive } as TypedField;
+    },
+  );
 
   //
   // Get object list.
   //
-  const objectList = paginatedResults.results.map((zaak) => ({
-    ...zaak,
-    href: formatMessage(REACT_APP_ZAAK_URL_TEMPLATE || "", zaak),
-  })) as unknown as AttributeData[];
+  const objectList = paginatedResults.results.map((zaak) => {
+    return {
+      ...zaak,
+      href: formatMessage(REACT_APP_ZAAK_URL_TEMPLATE || "", zaak),
+      acties: (
+        <>
+          {actions?.map(
+            ({ onClick, onMouseEnter, tooltip, ...action }, index) => {
+              const ButtonComponent = (
```

Comment:
can we move this into a separate function to keep this a little bit more readable?